### PR TITLE
feat: dark-theme

### PR DESCRIPTION
**Pull request by FigueiredoFelipe**
online

### CODE REVIEW

File: src/app/Providers.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+"use client";
```
Comment (jphilipstevens) [inline src/app/Providers.tsx:1]: what does use client do?

File: src/app/Providers.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import { ThemeProvider } from "next-themes";
+
+const Providers = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <ThemeProvider enableSystem={true} attribute="class">
+      <>{children}</>
```
Comment (jphilipstevens) [inline src/app/Providers.tsx]: I feel like i have done this but without needing <> </> around children 🤔

File: src/app/ThemeSwitcher.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    return null;
+  }
+
+  const handleThemeToggle = () => {
+    setTheme(theme === "light" ? "dark" : "light");
```
Comment (jphilipstevens) [inline src/app/ThemeSwitcher.tsx:20]: "light" and "dark" can be constants and not hard coded

File: src/app/layout.tsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import "./globals.css";
 import { Inter } from "next/font/google";
 import { Poppins } from "next/font/google";
+import Providers from "./Providers";
+// import ThemeSwitcher from "./ThemeSwitcher";
```
Comment (jphilipstevens) [inline src/app/layout.tsx:5]: remove comments out code

File: src/app/layout.tsx [hunk 5]
```diff
@@ -13,7 +15,6 @@ const poppins = Poppins({
 
 export const metadata = {
   title: "Felipe Figueiredo Portfolio",
-  // description: "Generated by create next app",
```
Comment (jphilipstevens) [inline src/app/layout.tsx:16]: same here remove this

**Review by jphilipstevens — APPROVED** [review]

File: src/app/Providers.tsx (hunk 1 above)
Comment (FigueiredoFelipe) [reply src/app/Providers.tsx:1]: By default, Next.js supports both server-side and client-side rendering, and you can access and manipulate DOM elements using standard React methods. The next/dynamic package allows for dynamic switching between server-side and client-side rendering if you have specific requirements that require client-side execution. **That line make it run in the client side to read the user currently dark/light themes.**

File: src/app/ThemeSwitcher.tsx (hunk 3 above)
Comment (FigueiredoFelipe) [reply src/app/ThemeSwitcher.tsx:20]: The themes 'light' and 'dark' are not hardcoded values. To handle the theme toggle functionality, the code imports the 'next-themes' dependency. This dependency provides utilities and features for handling themes in Next.js applications. One of the features it offers is the ability to retrieve the current browser theme.

**Pull request merged**